server/src/test/java/org/opensearch/index/seqno/ReplicationTrackerTests.java

```diff
@@ -1214,7 +1214,9 @@ public void testPeerRecoveryRetentionLeaseCreationAndRenewal() {
         IndexShardRoutingTable.Builder routingTableBuilder = new IndexShardRoutingTable.Builder(routingTable);
         for (ShardRouting replicaShard : routingTable.replicaShards()) {
             routingTableBuilder.removeShard(replicaShard);
-            routingTableBuilder.addShard(replicaShard.moveToStarted());
+            if (replicaShard.assignedToNode()) {
```

Comment:
I'd expect this to impact other tests, does it not?

Reply:
In all of the other tests, the replica shards are always assigned to a node, so the existing tests are passing.